### PR TITLE
[12.0][MIG] mass_editing: Migration to 12.0

### PR DESCRIPTION
**Pull request by aitorbouzas**
Migration to 12.0

### CODE REVIEW

**Comment by aitorbouzas** [comment]
I'll throw two questions:

1. Why isn't travis running?
2. When I run my tests locally [this line](https://github.com/OCA/server-ux/pull/29/files#diff-b0d63fd5b8ada4cfacc32861676faea3R263) fails because the uninstall hook cannot find the register in database (looks like there's not in there, is it because it is a transaction and it is not commited?)

**Comment by aitorbouzas** [comment]
Just fixed point 2 by deleting the actions with the env instead of a direct SQL execution with cr.

**Comment by aitorbouzas** [comment]
@pedrobaeza this is ready, but i'm getting lint errors with module date_range. What should I do?

**Comment by aitorbouzas** [comment]
Green and ready. @SerpentCS can you review?

File: mass_editing/README.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+* Go to *Settings / Mass Editing / Mass Editing* and configure the object and fields for Mass Editing.
+
+Usage
+=====
+
+This module allows to add, update or remove the values of more than one records on the fly at the same time.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/250/11.0
```
Comment (hugho-ad) [inline mass_editing/README.rst:56]: `s/11.0/12.0/g`

**Review by hugho-ad — APPROVED** [review]

File: mass_editing/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2016 Serpent Consulting Services Pvt. Ltd. (support@serpentcs.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Mass Editing',
+    'version': '12.0.1.0.0',
+    'author': 'Serpent Consulting Services Pvt. Ltd., '
+              'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'category': 'Tools',
+    'website': 'http://www.serpentcs.com',
```
Comment (simahawk) [inline mass_editing/__manifest__.py:10]: https://github.com/OCA/server-ux

File: mass_editing/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    'license': 'AGPL-3',
+    'summary': 'Mass Editing',
+    'uninstall_hook': 'uninstall_hook',
+    'depends': ['base'],
+    'data': [
+        'security/ir.model.access.csv',
+        'views/mass_editing_view.xml',
+    ],
+    'installable': True,
```
Comment (simahawk) [inline mass_editing/__manifest__.py:19]: all these flags are defaults, you can drop them

File: mass_editing/models/ir_model_fields.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    @api.model
+    def search(self, args, offset=0, limit=0, order=None, count=False):
+        model_domain = []
+        for domain in args:
+            if (len(domain) > 2 and domain[0] == 'mass_editing_domain' and
+                    isinstance(domain[2], str) and
+                    list(domain[2][1:-1])):
+                model_domain += [('model_id', 'in',
+                                  list(map(int, domain[2][1:-1].split(','))))]
```
Comment (simahawk) [inline mass_editing/models/ir_model_fields.py:18]: `[int(x) for x in domain[2][1:-1].split(',')]`

File: mass_editing/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,32 @@
+This module allows to add, update or remove the values of more than one records on the fly at the same time.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
```
Comment (simahawk) [inline mass_editing/readme/USAGE.rst:4]: remove this, it comes w/ README generator

File: mass_editing/tests/test_mass_editing.py [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo.tests import common
+from odoo.modules import registry
+from ..hooks import uninstall_hook
+
+
+class TestMassEditing(common.TransactionCase):
```
Comment (simahawk) [inline mass_editing/tests/test_mass_editing.py:12]: please boost test suite by using:
1. SavepointCase
2. setUp -> setUpClass
3. disable tracking `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

File: mass_editing/views/mass_editing_view.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                            widget="statinfo"/>
+                    </div>
+                    <div class="oe_title">
+                        <label for="name" class="oe_edit_only"/>
+                        <h1>
+                            <field name="name" required="1"/>
+                        </h1>
+                    </div>
+                    <group>
+                        <group>
```
Comment (simahawk) [inline mass_editing/views/mass_editing_view.xml:34]: name inner groups at least?

File: mass_editing/wizard/mass_editing_wizard.py [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        result =\
+            super(MassEditingWizard, self).fields_view_get(view_id=view_id,
+                                                           view_type=view_type,
+                                                           toolbar=toolbar,
+                                                           submenu=submenu)
+        context = self._context
```
Comment (simahawk) [inline mass_editing/wizard/mass_editing_wizard.py:22]: self.env.context

File: mass_editing/wizard/mass_editing_wizard.py [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+
+class MassEditingWizard(models.TransientModel):
+    _name = 'mass.editing.wizard'
+    _description = "Wizard for mass edition"
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        result =\
+            super(MassEditingWizard, self).fields_view_get(view_id=view_id,
```
Comment (simahawk) [inline mass_editing/wizard/mass_editing_wizard.py:18]: super()

File: mass_editing/wizard/mass_editing_wizard.py [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+                'special': 'cancel',
+            })
+            root = xml_form.getroottree()
+            result['arch'] = etree.tostring(root)
+            result['fields'] = all_fields
+        return result
+
+    @api.model
+    def create(self, vals):
+        if (self._context.get('active_model') and
```
Comment (simahawk) [inline mass_editing/wizard/mass_editing_wizard.py:237]: self.env.context

**Review by simahawk — CHANGES_REQUESTED** [review]

File: mass_editing/models/ir_model_fields.py (hunk 4 above)
Comment (aitorbouzas) [reply mass_editing/models/ir_model_fields.py:18]:
```suggestion
                                  [int(x) for x in domain[2][1:-1].split(',')]
```
Testing the new thingy

**Comment by hugho-ad** [comment]
I'll continue this, I'm interested to have this module migrated
https://github.com/OCA/server-ux/pull/41

**Pull request closed without merging**